File: Seccion 11 - Funciones en C++/Seccion_11_Dart/22_ejercicio_17.dart

import 'dart:io';

class Complejo {
  double real, imaginaria;

  Complejo();
}

Complejo z1 = new Complejo();
Complejo z2 = new Complejo();

void pedirDatos(Complejo z1, Complejo z2) {
  print('Digite los datos para el primer número complejo: ');
  stdout.writeln('Parte Real: ');
  z1.real = double.parse(stdin.readLineSync());
  stdout.writeln('Parte Imaginaria: ');
  z1.imaginaria = double.parse(stdin.readLineSync());

  print('Digite los datos para el segundo número complejo: ');
  stdout.writeln('Parte Real: ');
  z2.real = double.parse(stdin.readLineSync());
  stdout.writeln('Parte Imaginaria: ');
  z2.imaginaria = double.parse(stdin.readLineSync());
}

Complejo suma(Complejo z1, Complejo z2) {
  Complejo z = new Complejo();

  z.real = z1.real + z2.real;
  z.imaginaria = z1.imaginaria + z2.imaginaria;

  return z;
}

void muestra(Complejo x) {
  print('\n\nResultado de la suma: ${x.real} , ${x.imaginaria}');
}

void main() {

  pedirDatos(z1, z2);

  Complejo x = suma(z1, z2);

  muestra(x);
  
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/22_ejercicio_17.py

class Complejo:
    real = 0.0
    imaginaria = 0.0
    
    def __init__(self):
        pass

z1 = Complejo()
z2 = Complejo()

def pedir_datos():
    global z1, z2
    
    print('Digite los datos para el primer número complejo: ')
    z1.real = float(input('Parte Real: '))
    z1.imaginaria = float(input('Parte Imaginaria: '))
    
    print('Digite los datos para el segundo número complejo: ')
    z2.real = float(input('Parte Real: '))
    z2.imaginaria = float(input('Parte Imaginaria: '))

def suma(z1, z2):
    z = Complejo()
    
    z.real = z1.real + z2.real
    z.imaginaria = z1.imaginaria + z2.imaginaria
    
    return z

def muestra(x):
    print('\n\nEl resultado de la suma es: {} , {}'.format(x.real, x.imaginaria))

pedir_datos()

x = suma(z1, z2)

muestra(x)